src/modules/vtol_att_control/tiltrotor.cpp

```diff
@@ -266,35 +267,91 @@ void Tiltrotor::update_external_state()
-	_actuators_out_0->control[2] = _actuators_mc_in->control[2] * _roll_weight_mc;	// yaw
+	switch(_vtol_schedule.flight_mode) {
+		case MC_MODE:
+			_actuators_out_0->control[0] = _actuators_mc_in->control[0];
+			_actuators_out_0->control[1] = _actuators_mc_in->control[1];
+			_actuators_out_0->control[2] = _actuators_mc_in->control[2];
+			_actuators_out_0->control[3] = _actuators_mc_in->control[3];
+			_actuators_out_1->control[0] = 0;
+			_actuators_out_1->control[1] = 0;
+			_actuators_out_1->control[4] = _tilt_control;
+			break;
+		case FW_MODE:
+			_actuators_out_0->control[0] = 0;
+			_actuators_out_0->control[1] = 0;
+			_actuators_out_0->control[2] = 0;
+			_actuators_out_0->control[3] = _actuators_fw_in->control[3];
+
+			_actuators_out_1->control[0] = -_actuators_fw_in->control[0];	// roll elevon
+			_actuators_out_1->control[1] = _actuators_fw_in->control[1] + _params->fw_pitch_trim;	// pitch elevon
+			_actuators_out_1->control[2] = _actuators_fw_in->control[2];	// yaw
+			_actuators_out_1->control[3] = _actuators_fw_in->control[3];	// throttle
+			_actuators_out_1->control[4] = _tilt_control;
+			break;
+		case TRANSITION_FRONT_P1:
+			_actuators_out_0->control[0] = _actuators_mc_in->control[0] * _roll_weight_mc;
+			_actuators_out_0->control[1] = _actuators_mc_in->control[1];
+			_actuators_out_0->control[2] = _actuators_mc_in->control[2] * _yaw_weight_mc;
+			_actuators_out_0->control[3] = _actuators_mc_in->control[3];
+			_actuators_out_1->control[0] = -_actuators_fw_in->control[0] * (1.0f - _roll_weight_mc);	//roll elevon
+			_actuators_out_1->control[1] = (_actuators_fw_in->control[1] + _params->fw_pitch_trim);		//pitch elevon
+			_actuators_out_1->control[4] = _tilt_control;	// for tilt-rotor control
+			break;
+		case TRANSITION_FRONT_P2:
+			_actuators_out_0->control[0] = 0;
+			_actuators_out_0->control[1] = 0;
+			_actuators_out_0->control[2] = 0;
+			_actuators_out_0->control[3] = _actuators_fw_in->control[3];
 
-	if (_vtol_schedule.flight_mode == FW_MODE) {
-		_actuators_out_1->control[3] = _actuators_fw_in->control[3];	// fw throttle
-	} else {
-		_actuators_out_0->control[3] = _actuators_mc_in->control[3];	// mc throttle
-	}
+			_actuators_out_1->control[0] = -_actuators_fw_in->control[0];	// roll elevon
+			_actuators_out_1->control[1] = _actuators_fw_in->control[1] + _params->fw_pitch_trim;	// pitch elevon
+			_actuators_out_1->control[2] = _actuators_fw_in->control[2];	// yaw
+			_actuators_out_1->control[3] = _actuators_fw_in->control[3];	// throttle
+			_actuators_out_1->control[4] = _tilt_control;					// tilt
 
-	_actuators_out_1->control[0] = -_actuators_fw_in->control[0] * (1.0f - _roll_weight_mc);	//roll elevon
-	_actuators_out_1->control[1] = (_actuators_fw_in->control[1] + _params->fw_pitch_trim)* (1.0f -_roll_weight_mc);	//pitch elevon	
-	_actuators_out_1->control[4] = _tilt_control;	// for tilt-rotor control
+			break;
+		case TRANSITION_BACK:
```

Comment:
This is the same as TRANSITION_FRONT_P1 with the exception of roll_weight_mc. Intentional?